web/src/pages/your_account.rs

```diff
@@ -0,0 +1,299 @@
+        }
+    }
+
+    Some(amount)
+}
+
+impl Component for YourAccount {
+    type Properties = Properties;
+    type Message = Message;
+
+    fn create(props: Properties, link: ComponentLink<Self>) -> Self {
+        let send_asset_agent = agents::SendAsset::bridge(link.callback(Message::TransactionSent));
+        let get_latest_transactions_agent =
+            agents::GetLatestTransactions::bridge(link.callback(Message::LatestTransactionsGot));
+        let get_users_agent = agents::GetUsers::bridge(link.callback(Message::GotUsers));
+
+        let mut get_balance_agent = agents::GetBalance::bridge(link.callback(Message::GotBalance));
+        get_balance_agent.send(props.user.0.clone().to_thin());
+
+        let refresh = link.callback(|_: ()| Message::Refresh);
+
+        Self {
+            link,
+            props,
+
+            get_balance_agent,
+            user_balance: None,
+
+            send_asset_agent,
+            user_to_send_to: None,
+            amount_to_send: "3".to_owned(),
+
+            get_latest_transactions_agent,
+            latest_transactions: Vec::new(),
+            get_users_agent,
+            pubkey_to_username: HashMap::new(),
+            refresher: Interval::new(1000, move || {
+                refresh.emit(());
+            }),
+        }
+    }
+
+    fn update(&mut self, message: Message) -> ShouldRender {
+        match message {
+            Message::GotBalance(res) => {
+                match res {
+                    Ok(balance) => self.user_balance = Some(balance),
+                    Err(_) => self
+                        .get_balance_agent
+                        .send(self.props.user.0.clone().to_thin()),
+                }
+                true
+            }
+
+            Message::SelectUser(user) => {
+                self.user_to_send_to = Some(user);
+                false
+            }
+            Message::UpdateAmount(amount) => {
+                self.amount_to_send = amount;
+                true
+            }
+            Message::SendTransaction => {
+                if let Some(amount) = validate_amount(&self.user_balance, &self.amount_to_send) {
+                    if let Some(user_to_send_to) = self.user_to_send_to.clone() {
+                        let sequence = self.props.user.1 + 1;
+
+                        self.send_asset_agent.send((
+                            self.props.user.0.clone(),
+                            sequence,
+                            user_to_send_to,
+                            amount,
+                        ));
+
+                        self.props.bump_sequence.emit(sequence);
+                    }
+                }
+
+                // TODO in background
+                self.get_balance_agent
+                    .send(self.props.user.0.clone().to_thin());
+
+                false
+            }
+            Message::TransactionSent(ret) => {
+                ret.unwrap(); // TODO send asset in dialog
```

Comment:
important todo? Is it OK to `unwrap` here?

Reply:
Dangling from dialog, but yeah, let's not `unwrap`, the sent tx appears in the txs list anyway,